scripts/config-build.py

```diff
@@ -206,14 +206,14 @@ def main(calling_script, args, unknown_args):
         cmake_line.append("--graphviz=dependency.dot")
         dot_line = "dot -Tpng dependency.dot -o dependency.png"
 
-    # Append cache file at the end of the command line to make previous argument visible to the cache.
-    cmake_line.append("-C%s" % cache_file)
-
     for unknown_arg in unknown_args:
         if not unknown_arg.startswith('-D'):
             logging.warning("Additional argument '%s' does not start with '-D'. Keeping it nevertheless." % unknown_arg)
         cmake_line.append(unknown_arg)
 
+    # Append cache file at the end of the command line to make previous argument visible to the cache.
+    cmake_line.append("-C%s" % cache_file)
```

Comment:
@victorapm Why did you put this line before the `-D` arguments in your PR?
Was there something that you needed done?

Reply:
Oops! This change was not supposed to get pushed yet, sorry for doing that!

At some point, I was experimenting with oversubscribing cmake options that were already defined in the `.cmake` file (HYPRE_DIR to be more precise). So I flipped the append order of `cmake_line` to make that possible. But clearly I wasn't care of side effects...

Do you believe we could have this behavior with the build script? i.e., over-subscribe options defined on the `.cmake` file by default when they are passed via command line to the build script?

Reply:
My take it that this script is supposed to be an entry point that wraps `cmake` (which can be complicated to tame if you're not used to it regularly).

If you need something really specific witch `cmake`, I'd recommend that you consider getting the information from the `cmake_cmd` file, generated by `config-build.py`.

For example, on my side, I never use this python script, but `cmake` directly.

Reply:
Good point! Using cmake directly is probably the best way to go in the situation I described

Reply:
@victorapm , You can have test if the variable is already defined in your host-config file. This is what I do to check user gave GEOSX_TPL_DIR on Pangea3 and what was broken when the line is moved. Not sure I answer your question correctly, maybe I misunderstood.

Reply:
Thanks, all!